### ISSUE TITLE
False positive DAR101 (missing parameter) for __init_subclass__

### ISSUE THREAD

**Issue by Dominik1123** [terrencepreilly/darglint#100]
According to the [Python Data Model](https://docs.python.org/3/reference/datamodel.html#object.__init_subclass__) about `__init_subclass__`:

> [...] If defined as a normal instance method, this method is implicitly converted to a class method.

However darglint reports a missing parameter `cls` in this case. For example:

```python
class Test:
    def __init_subclass__(cls, foo):
        """Do something.

        Args:
            foo: Do something with foo.
        """
        pass
```

**Comment by ValentinCalomme** [comment]
I found that adding the `@classmethod` decorator suppresses that error. The following should work.

```python
class Test:
    @classmethod
    def __init_subclass__(cls, foo):
        """Do something.

        Args:
            foo: Do something with foo.
        """
        pass
```

**Comment by Dominik1123** [comment]
That's true, but according to the Python specs `__init_subclass__` does not
need to be decorated by `classmethod` as it will be implicitly converted.

Decorating is a workaround for now but I don't like adding extra noise to
my code only to satisfy the linter. Instead darglint should special-case
that method to recognize it as a classmethod no matter if decorated or not.
